src/common/ceph_hash.cc

```diff
@@ -82,7 +82,7 @@ unsigned ceph_str_hash_rjenkins(const char *str, unsigned length)
  */
 unsigned ceph_str_hash_linux(const char *str, unsigned length)
 {
-	unsigned long hash = 0;
+	unsigned hash = 0;
```

Comment:
thanks tchaikov.

i tried "uint_fast32_t", the performance is same as old code, and i found it's defined as "long int" on 64-bit platform(/usr/include/stdint.h). from compiler's point of view, it makes sense as integer with register width in length can be handled efficiently. but it just isn't smart enough to tune all cases.

about "change return type to uint32_t", i think it's necessary as the old code uses "non-type unsigned" extensively which is not clear enough. because there are many codes in this file using "non-type unsigned" and they're related to each other, maybe it's better to push another commit to fix this issue specifically?